### DISCUSSION TITLE
How to get list of guild members?

### DISCUSSION THREAD

**Discussion by DeedsBaron** [bwmarrin/discordgo#1135] [Q&A] [answered]
Hello i'm stuck with this point, How can i get a list of members of the guild?

**Comment by DeedsBaron** [comment]
I've already enabled guild intent on Discord site, and done `	goBot.Identify.Intents = discordgo.MakeIntent(discordgo.IntentsAll)
`. Trying to recieve Members slice by `guild.Members` but im recieving empty slice(

**Comment by FedorLap2006** [comment, marked as answer]
Only state (retrieved through methods of `session.State`) guilds will have `Members` property set. Instances returned from the API will return an empty slice.
You can request all guild members however, using `session.GuildMembersList` method.

**Reply by DeedsBaron** [reply]
Much thanks!

**Reply by sohneg** [reply]
> Only state (retrieved through methods of `session.State`) guilds will have `Members` property set. Instances returned from the API will return an empty slice. You can request all guild members however, using `session.GuildMembersList` method.

How can i accces this session.GuildMembersList method? I would like to add all members in a db when creating the session first time. My code looks like this right now:

```
func CreateSession(s *discordgo.Session) {
	// Create a new Discord Session
	dg, err := discordgo.New("Bot " + token)
	if err != nil {
		fmt.Println("Error creating Discord session, ", err)
		return
	}
	// Register messageCreate as callback for MessageCreate events
	dg.AddHandler(messageCreate)
	// Care about receiving messages
	dg.Identify.Intents = discordgo.IntentGuildMessages


	// Open websocket to Discord and begin listening
	err = dg.Open()
	if err != nil {
		fmt.Println("Error opening connection, ", err)
	}

	fmt.Println("Bot is now running. Press CTRL-C to exit")
	sc := make(chan os.Signal, 1)
	signal.Notify(sc, syscall.SIGINT, syscall.SIGTERM, os.Interrupt)
	<-sc
	dg.Close()
}
```

**Comment by Cypaaa** [comment]
Hi, in the current last version (0.26.1) there is no `GuildMembersList` and s.State.Guilds[n].Members keep retruning an empty list.
**But** we can get a member list with [GuildMembers](https://pkg.go.dev/github.com/bwmarrin/discordgo@v0.26.1#Session.GuildMembers) but it returns a list with __a limit of 1000 members__.

So, is there any way in this version (0.26.1) to get a list of all members without a limit, I'm not talking on looping over a method to complete the list.

Thanks for your answers ! :)

**Reply by FedorLap2006** [reply]
I believe `RequestGuildMembersList` is what you seek. It was not removed, just has a different name.

**Reply by Cypaaa** [reply]
My problem with this method is that it require me to pass many values when i just want to get the members of a guild without having to pass any defined values like their id ect.
I assume that the signature` func (s *Session) RequestGuildMembersList(guildID string, userIDs []string, limit int, nonce string, presences bool) error` tells use we have to send a list of userIDs and I don't really understand why (had a rough day, didn't had the time to check the code of the method).
But yeah thanks, i'll try to send an empty list to see what it gives me.

**Reply by Cypaaa** [reply]
yeah i get an empty GuildMembersChunk, only the guild ID is set, I might not use it properly then

**Reply by FedorLap2006** [reply]
> tells use we have to send a list of userIDs and I don't really understand why

Oh, that's right, my bad. There are two such functions:
- `RequestGuildMembersList` - request certain guild members
- `RequestGuildMembers` - request all guild members

You need the latter.

> yeah i get an empty GuildMembersChunk, only the guild ID is set, I might not use it properly then

This is due to it member chunks being a privileged endpoint, i.e. requiring you to have a `GUILD_MEMBERS` intent.
